app/classes/query/modules/google_search.rb

```diff
@@ -70,13 +70,13 @@ def google_conditions(search, field)
 
   # Pull off one "and" clause from the beginning of the string.
   def google_parse_one_clause(str, goods, bads)
-    if str.sub!(/^-"([^""]+)"( |$)/, "") ||
+    if str.sub!(/^-"([^"]+)"( |$)/, "") ||
        str.sub!(/^-(\S+)( |$)/, "")
       bads << Regexp.last_match(1)
-    elsif str.sub!(/^(("[^""]+"|\S+)( OR ("[^""]+"|\S+))*)( |$)/, "")
+    elsif str.sub!(/^(("[^"]+"|\S+)( OR ("[^"]+"|\S+))*)( |$)/, "")
       str2 = Regexp.last_match(1)
       or_strs = []
-      while str2.sub!(/^"([^""]+)"( OR |$)/, "") ||
+      while str2.sub!(/^"([^"]+)"( OR |$)/, "") ||
```

Comment:
Autocorrected by Lint/DuplicateRegexpCharacterClassElement cop
https://docs.rubocop.org/rubocop/1.31/cops_lint.html#lintduplicateregexpcharacterclasselement

Reply:
The reason I (occasionally remember to) do the redundant double quotes is that it sometimes works better with syntax highlighters, especially when there is a typo somewhere which causes the parser to go off the rails.

A better example of the same thing (which might not offend rubocop) is to avoid unbalanced parentheses, brackets or braces, adding the closing paren even if unnecessary, or if that would break the expression, adding the closing paren in a comment.  In vim there is a command to jump between the opening and closing parens surrounding the cursor.  That breaks if you have a string literal with just one paren "(".

I take it the syntax highlighters you and Nimmo use will not be fooled by the unbalanced double quotes in some of these lines?  If so, then I have no qualms about this particular example, personally.  (I don't even like or use syntax highlighting!)

Reply:
@pellaea: Thanks for reviewing yet another PR so carefully.
Good point about syntax highlighting; I hadn't thought of that.
Fortunately, VSCode (which both @nimmolo and I use) isn't fooled by the changes. Or at least the changes don't make any difference in the display.